### PR TITLE
feat: SSO MFA - WebUI backend implementation

### PR DESCRIPTION
**Pull request by Joerger**
Part of the implementation of https://github.com/gravitational/teleport/pull/44699

Add SSO MFA challenges and response to the web http api and websocket messages.

Depends on https://github.com/gravitational/teleport/pull/46982

frontend implementation (per-session mfa only): https://github.com/gravitational/teleport/pull/47876

### CODE REVIEW

**Comment by Joerger** [comment]
@Reviewers spooky ping to review 👻

**Comment by Joerger** [comment]
@EdwardDowling @probakowski @rosstimothy Friendly ping to review

**Review by probakowski — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

File: lib/client/weblogin.go [hunk 1]
```diff
@@ -111,6 +111,14 @@ type MFAChallengeResponse struct {
 	TOTPCode string `json:"totp_code,omitempty"`
 	// WebauthnResponse is a response from a webauthn device.
 	WebauthnResponse *wantypes.CredentialAssertionResponse `json:"webauthn_response,omitempty"`
+	// SSOResponse is a response from an SSO MFA flow.
+	SSOResponse *SSOResponse `json:"sso_response"`
+}
+
+// SSOResponse is a json compatible [proto.SSOResponse].
+type SSOResponse struct {
+	RequestID string `json:"requestId,omitempty"`
```
[the remark this answers is not in the document]
Comment (Joerger) [reply lib/client/weblogin.go:120]: This is intentional, the frontend uses camel case, but we've been inconsistent with it on the backend.

[the remark this answers is not in the document]
Comment (Joerger) [reply lib/client/weblogin.go:120]: See a bit lower:

```go
type AuthenticateWebUserRequest struct {
	// User is a teleport username.
	User string `json:"user"`
	// WebauthnAssertionResponse is a signed WebAuthn credential assertion.
	WebauthnAssertionResponse *wantypes.CredentialAssertionResponse `json:"webauthnAssertionResponse,omitempty"`
}
```

File: lib/client/weblogin.go [hunk 2]
```diff
@@ -457,6 +465,37 @@ type MFAAuthenticateChallenge struct {
...
+	// MFA requests so they don't interfere with each other.
+	ChannelID string `json:"channelId"`
+}
+
+// SSOMFADevice is a json compatible [proto.SSOMFADevice].
+type SSOMFADevice struct {
+	ConnectorID   string `json:"connectorId,omitempty"`
+	ConnectorType string `json:"connectorType,omitempty"`
+	DisplayName   string `json:"displayName,omitempty"`
+}
```
[the remark this answers is not in the document]
Comment (Joerger) [reply lib/client/weblogin.go:487]: Same as above

**Pull request merged**